### PR TITLE
workload/kv: add --secondary-index flag

### PR DESCRIPTION
**Pull request by nvb**
The flag instructs schema initialization to add a secondary index
on the `v` column. This has been useful for experimenting with the
perf impact of secondary indexes.

Release note: None

### CODE REVIEW

File: pkg/workload/kv/kv.go [hunk 1]
```diff
@@ -121,6 +125,13 @@ func (w *kv) Hooks() workload.Hooks {
 			}
 			return nil
 		},
+		PostLoad: func(sqlDB *gosql.DB) error {
+			if w.secondaryIndex {
+				_, err := sqlDB.Exec(`CREATE INDEX IF NOT EXISTS v_idx ON kv (v)`)
```
Comment (tschottdorf) [inline pkg/workload/kv/kv.go]: Is this going to be backfilled in the background? If so, the impact of that will be pretty nondeterministic. Maybe that doesn't matter now, but perhaps call it out on the flag.

**Review by tbg — APPROVED** [review]

**Review by a-robinson — APPROVED** [review]
LGTM, although I'm also curious about the fact that creating the index is done `PostLoad` rather than `PreLoad`. I assume it's just meant to rely on the fact that we don't use fixtures for `kv`?

**Comment by nvanbenschoten** [comment]
Both questions here are good and I had similar ones while making the change. Putting this in `PostLoad` was inspired by `workload/tpcc`. @danhhz is that how you would expect this to work? An alternative would be to change the `workload.Table.Schema` field based on the flag.

File: pkg/workload/kv/kv.go (hunk 1 above)
Comment (danhhz) [inline pkg/workload/kv/kv.go]: if you keep this as PostLoad, it should be marked as a runtime-only flag, since it doesn't affect fixtures here

**Review by danhhz — COMMENTED** [review]
I would advocate for changing the schema based on the flag so it will work better with fixtures. tpcc adds foreign keys in `PostLoad`, but it's very careful to make the indexes needed for those foreign keys ahead of time in the schema (which means they're in the fixtures).

**Review by danhhz — APPROVED** [review]

**Comment by nvanbenschoten** [comment]
TFTRs.

bors r+

**Review by a-robinson — APPROVED** [review]
Nice, LGTM.

**Pull request merged**